pkg/gds/interceptor.go

```diff
@@ -26,3 +42,71 @@ func (s *Service) serverInterceptor(ctx context.Context, in interface{}, info *g
 	log.Debug().Str("method", info.FullMethod).Str("latency", time.Since(start).String()).Err(err)
 	return out, err
 }
+
+func (s *Service) replicaInterceptor(ctx context.Context, in interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (out interface{}, err error) {
+	// Track how long the method takes to execute.
+	start := time.Now()
+
+	// Check if we're in maintenance mode - status method should still return a full response.
+	if s.conf.Maintenance && info.FullMethod != "/trisa.gds.api.v1beta1.TRISADirectory/Status" {
+		return nil, status.Error(codes.Unavailable, "the GDS service is currently in maintenance mode")
+	}
+
+	// Fetch peer information from the TLS info.
+	var peer *PeerInfo
+	if peer, err = peerFromTLS(ctx); err != nil {
+		log.Error().Err(err).Msg("unable to retrieve remote peer info")
+		// TODO: Uncomment when mTLS is added
+		//return nil, status.Error(codes.Unauthenticated, "unable to retrieve authenticated peer information")
```

Comment:
Perfect!

Reply:
When we add mTLS are there any other checks we need to do to ensure mTLS is valid?